boto/connection.py

```diff
@@ -793,18 +793,21 @@ def proxy_ssl(self, host=None, port=None):
         else:
             sock = socket.create_connection((self.proxy, int(self.proxy_port)))
         boto.log.debug("Proxy connection: CONNECT %s HTTP/1.0\r\n", host)
-        sock.sendall("CONNECT %s HTTP/1.0\r\n" % host)
-        sock.sendall("User-Agent: %s\r\n" % UserAgent)
+        sock.sendall(("CONNECT %s HTTP/1.0\r\n" % host).encode())
```

Comment:
You might want to explicitly say encode("UTF-8")
That's the default under py3, but py2 it depends on sys.getdefaultencoding().
It's likely ok the way it is, but it provides consistency.

Reply:
Strictly speaking, non-ASCII hostnames should be "idna"-encoded, rather than UTF-8 or a system's default, as IDNA is the standard for internationalized domain names.